boards/arm/qemu_cortex_m3/Kconfig.defconfig

```diff
@@ -12,4 +12,10 @@ config GPIO_STELLARIS
 	default y
 	depends on GPIO
 
+# DWT is not properly emulated in QEMU
+choice CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION
+	bool
+	default CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_NONE
```

Comment:
ACAICT this doesn't prevent a user from enabling DETECTION via menuconfig. Could this instead be a `depends on !BOARD_QEMU_CORTEX_M3` in the `CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_*` entries?

Reply:
Thanks @oyvindronningstad, this is correct, although, I don't see why one would like to do this for qemu_cortex_m3, which is _only_ a QEMU emulation board, not used on real boards.

Ideally, the QEMU_CORTEX_M3 board should not have CPU_HAS_DWT selected, but since the corresponding SoC does have DWT, there is no ideal solution.

Reply:
Adding a depends on !BOARD_QEMU_CORTEX_M3 in the arch-level definitions is not beautiful at all, I would prefer to live with the issue you're raising instead of polluting the arch Kconfig files with a workaround...

Reply:
This won't ever break CI. The only problem is if anyone _manually_ wants to run an application on qemu_cortex_m3 with Null-pointer detection enabled in prj.conf.

Reply:
Ok. This isn't a huge problem.